app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -242,57 +246,72 @@ async function addEthereumChainHandler(
     );
   }
   let networkConfigurationId;
+
+  const { id: approvalFlowId } = await startApprovalFlow();
+
   try {
-    await requestUserApproval({
-      origin,
-      type: ApprovalType.AddEthereumChain,
-      requestData: {
-        chainId: _chainId,
-        rpcPrefs: { blockExplorerUrl: firstValidBlockExplorerUrl },
-        chainName: _chainName,
-        rpcUrl: firstValidRPCUrl,
-        ticker,
-      },
-    });
-
-    networkConfigurationId = await upsertNetworkConfiguration(
-      {
-        chainId: _chainId,
-        rpcPrefs: { blockExplorerUrl: firstValidBlockExplorerUrl },
-        nickname: _chainName,
-        rpcUrl: firstValidRPCUrl,
-        ticker,
-      },
-      { source: MetaMetricsNetworkEventSource.Dapp, referrer: origin },
-    );
+    try {
```

Comment:
Very minor, but is it worth splitting this big block into `addEthereumChain` and `switchEthereumChain` methods to avoid the nesting of multiple `try catch`?

Reply:
If the problem is the nested try/catch, then I'd rather remove the top level try/finally I added and, instead, repeat `endApprovalFlow({ id: approvalFlowId });` in any `try/catch` that requires it.

The problem with moving those blocks into functions is that those blocks sometimes finish execution and sometimes they don't, so even if you move them to a function you still need to return an error to process it, or to throw an error and still add the try/catch.

I've committed the changes with the top level try/finally gone so that you can see how it looks, I'm pretty ambivalent about either option.

Reply:
Good point, in that case I'm happy with either so lets leave with a single `try catch` and multiple `endApprovalFlow`  for the sake of readability.